split-gpg2-client

```diff
@@ -1,21 +1,35 @@
-#!/bin/bash
+#!/bin/bash --
+set -euo pipefail
 
 for d in /etc "${XDG_CONFIG_HOME:-$HOME/.config}"; do
     rc_file="$d/split-gpg2-rc"
-    if [ -r "$rc_file" ]; then
+    if [[ -r "$rc_file" ]]; then
         . "$rc_file"
     fi
 done
 
-if [[ -z "$SPLIT_GPG2_SERVER_DOMAIN" ]]; then
+if ! [[ -v SPLIT_GPG2_SERVER_DOMAIN ]]; then
     SPLIT_GPG2_SERVER_DOMAIN="@default"
+elif [[ "$SPLIT_GPG2_SERVER_DOMAIN" =~ ^[A-Za-z@][0-9A-Za-z_-]*$ ]]; then
+    :
+else
+    # prevent accidental command injection into socat
+    printf '$SPLIT_GPG2_SERVER_DOMAIN (%q) is not a valid qrexec target\n' "$SPLIT_GPG2_SERVER_DOMAIN" >&2
+    exit 1
 fi
 
-agent_socket="$(gpgconf --list-dirs | grep '^agent-socket:' | cut -d ':' -f 2)"
-if [[ -z "$agent_socket" ]]; then
-    echo "Failed to find gpg-agent socket path" >&2
+agent_socket=$(gpgconf --list-dirs -o/dev/stdout |
+    LC_ALL=C awk -F: '/^agent-socket:/ { print $2 }')
+
+# Again, socat struggles with metacharacters, so reject them
+if [[ $agent_socket =~ ^/[A-Za-z0-9/+_.-]+$ ]]; then
+    exec socat "unix-listen:'$agent_socket',fork,unlink-early" \
+        "exec:qrexec-client-vm $SPLIT_GPG2_SERVER_DOMAIN qubes.Gpg2"
```

Comment:
I think validating the trusted `SPLIT_GPG2_SERVER_DOMAIN` and `agent_socket` for socat metachars is not worth the clutter it adds. So unless you have an realistic scenario where it not just breaks split-gpg so the user notice their error I won't take this change.

Reply:
Even if it just breaks split-gpg, I still strongly prefer up-front validation, as it results in better error messages.  “not a valid qrexec target” is much more helpful than “Request refused”, for example.  `agent_socket` validation can be folded into the grep or awk regular expression, which removes most of the clutter.

Reply:
To elaborate: Validating SPLIT_GPG2_SERVER_DOMAIN up-front allows for a much clearer error message.  Otherwise, a malformed value would likely result in error messages that have very little to do with what the problem actually was.